scripts/include/helpers.inc.php

```diff
@@ -302,6 +302,45 @@ function formatXliffForWeblate($file, $setApproved = false)
   file_put_contents($file, $xml);
 }
 
+// Weblate will not preserve leading and trailing blank spaces or newlines by default.
+// Set xml:space="preserve" attribute on source and target strings to override this 
+// behaviour.
+function setPreserveSpacing($file)
+{
+  $changed = 0;
+
+  $xml = simplexml_load_file($file);
+
+  $elementList = ["//source", "//target"];
+
+  foreach ($elementList as $elementName) {
+    $elements = $xml->xpath($elementName);
+
+    foreach ($elements as $element) {
+      // Get the xml namespace attributes for this element.
+      $xmlnsAttrs = [];
+      if (isset($element->getNamespaces(true)['xml'])) {
+        $xmlnsAttrs = $element->attributes($element->getNamespaces(true)['xml']);
+      }
+
+      if (isset($xmlnsAttrs['space']) && strtolower($xmlnsAttrs['space']) != 'preserve') {
+        $xmlnsAttrs['space'] = 'preserve';
```

Comment:
Hmm, don't you need to update the `$xml` object representing the DOM with the new "space=preserve" value, or is it sufficient to change the value in `$xmlnsAttrs` for `$xml` to get updated?

Reply:
$xml is a [SimpleXMLElement](https://www.php.net/manual/en/class.simplexmlelement.php) which acts more like a resource than an object. All of it's methods return references to SimpleXMLElement types. This code functions to add the attr to the output xml.